### PR TITLE
Delete only empty namespace

### PR DESCRIPTION
**Pull request by mhmxs**
${title}

### CODE REVIEW

File: pkg/installer/uninstall.go [hunk 1]
```diff
@@ -2,14 +2,24 @@ package installer
...
 
 	apiv1 "github.com/storageos/kubectl-storageos/api/v1"
 	pluginutils "github.com/storageos/kubectl-storageos/pkg/utils"
-	"k8s.io/apimachinery/pkg/api/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	"sigs.k8s.io/kustomize/api/krusty"
 )
 
+const skipNamespaceDeletionMessage = `Namespace %s still has resources.
+	Deletion of namespace has skipped.
```
Comment (croomes) [inline pkg/installer/uninstall.go:16]: Nit: "Deletion of namespace has skipped" -> "Skipped namespace removal"

File: pkg/installer/uninstall.go [hunk 2]
```diff
@@ -2,14 +2,24 @@ package installer
...
+	Reason: %s
+	Please check the existing resources by executing the following command:
```
Comment (croomes) [inline pkg/installer/uninstall.go:18]: "Check for resources remaining in the namespace with:"

File: pkg/installer/uninstall.go [hunk 3]
```diff
@@ -2,14 +2,24 @@ package installer
...
+	kubectl get all -n %s
+	To delete namespace please execute the command below:
```
Comment (croomes) [inline pkg/installer/uninstall.go:20]: "To remove the namespace and all remaining resources within it, run:"

File: pkg/installer/uninstall.go [hunk 4]
```diff
@@ -225,25 +237,21 @@ func (in *Installer) kustomizeAndDelete(dir, file string) error {
 
 // gracefullyDeleteNS deletes a k8s namespace only once there are no pod running in said namespace,
```
Comment (croomes) [inline pkg/installer/uninstall.go:238]: Update comment "pod" -> "resources"

File: pkg/installer/uninstall.go [hunk 5]
```diff
@@ -225,25 +237,21 @@ func (in *Installer) kustomizeAndDelete(dir, file string) error {
...
+func (in *Installer) gracefullyDeleteNS(namespacename string) error {
+	err := pluginutils.WaitFor(func() error {
+		return pluginutils.NoResourcesInNS(in.clientConfig, namespacename)
+	}, 180, 10)
 	if err != nil {
 		return err
 	}
 
-	err = in.kubectlClient.Delete(context.TODO(), "", namespaceManifest, true)
+	err = pluginutils.DeleteNamespace(in.clientConfig, namespacename)
```
Comment (croomes) [inline pkg/installer/uninstall.go:248]: nit: I'd use (here and elsewhere):
```
if err := pluginutils.DeleteNamespace(in.clientConfig, namespacename); err != nil {
  return err
}
```

File: pkg/utils/k8s.go [hunk 6]
```diff
@@ -172,7 +190,7 @@ func WaitFor(fn func() error, limit, interval time.Duration) error {
 	for {
 		select {
 		case <-timeout:
-			return fmt.Errorf("timeout with error: %v", err)
+			return fmt.Errorf("timeout with error: %w", err)
```
Comment (croomes) [inline pkg/utils/k8s.go:193]: `errors.Wrap()`?

**Review by croomes — COMMENTED** [review]
Looks good, some minor nits.

File: pkg/utils/k8s.go (hunk 6 above)
Comment (mhmxs) [reply pkg/utils/k8s.go:193]: errors doesn't have `Wrap()` function, `%w` does it.

File: pkg/installer/uninstall.go (hunk 5 above)
Comment (mhmxs) [reply pkg/installer/uninstall.go:248]: I don't really understand. Other places we use resource manifest to delete, this is the only place where we delete by name.

**Comment by mhmxs** [comment]
Some words about `--skip-namespace-deletion`. So the default is false. That means in scripts uninstaller deletes the namespace by default. But if the flag has not set and the installer founds an interactive terminal it asks for confirmation. Otherwise uninstaller uses the passed or default value.

File: pkg/installer/uninstall.go (hunk 5 above)
Comment (croomes) [reply pkg/installer/uninstall.go]: It was just code style, when a function only returns an error you'd typically scope it to the if statement that checks it.  Not a big deal, I see it's done this way in most other places in the project.

File: pkg/utils/k8s.go (hunk 6 above)
Comment (croomes) [reply pkg/utils/k8s.go:192]: We started using `"github.com/pkg/errors"` in api-manager and operator-toolkit.  It does the same - it just seems a little more friendly.  It's very subjective of course...

Comment (mhmxs) [reply pkg/utils/k8s.go:192]: Thanks. I don't want to change all the places in this PR.

File: pkg/installer/uninstall.go (hunk 5 above)
Comment (mhmxs) [reply pkg/installer/uninstall.go]: The project doesn't follow this pattern. I will fix this one, because i also prefer.

Comment (croomes) [reply pkg/installer/uninstall.go]: yeah, I hadn't realised at the time that the project doesn't follow the pattern.  Not a big big deal, and better to be consistent.  Maybe we revisit later.

**Review by croomes — APPROVED** [review]
Looks good, thanks!

Can you check the tests though?

**Pull request merged**